APMrover2/mode_auto.cpp

```diff
@@ -131,6 +131,40 @@ float ModeAuto::get_distance_to_destination() const
     return 0.0f;
 }
 
+// get desired location
+bool ModeAuto::get_desired_location(Location& destination)
+{
+    switch (_submode) {
+    case Auto_WP:
+        if (g2.wp_nav.is_destination_valid()) {
+            destination = g2.wp_nav.get_destination();
+            return true;
+        } else {
+            return false;
+        }
+        break;
+    case Auto_HeadingAndSpeed:
+        // always return true because this is the safer option to allow missions to continue
+        return true;
+        break;
+    case Auto_RTL:
+        // get destination from RTL
+        return rover.mode_rtl.get_desired_location(destination);
+        break;
+    case Auto_Loiter:
+        // get destination from loiter
+        return rover.mode_loiter.get_desired_location(destination);
+        break;
+    case Auto_Guided:
+        // get destination from guided
+        return rover.mode_guided.get_desired_location(destination);
+        break;
+    }
+
+    // we should never reach here but just in case
```

Comment:
I've been considering adding an internal error for this "we really shouldn't be here" case.

Reply:
sure, sounds like a good idea..